src/vehicle.cpp

```diff
@@ -875,6 +875,13 @@ void vehicle::use_controls()
         menu.addentry( release_remote_control, true, 'l', _("Stop controlling") );
     }
 
+    bool has_electronic_controls = false;
+
+    for( size_t p = 0; p < parts.size(); p++ ) {
```

Comment:
You can use `part_with_feature` to search all parts on the same square as a given part `p` (returns a part index, so compare it via `>= 0`). But the real problem is probably not knowing the "current part" (the part that is examined and triggered the call to `use_controls`). That information is only known to the caller, and must be added to the function call.